refactor: remove old message files

juliapampus opened the pull request:
## What this PR changes/adds

Removes old schema and example files from specifications folder. Updates .puml files according to new schemas.

## Why it does that

The removed/changed files were outdated.

## Further notes

Will cause merge conflicts #104

## Linked Issue(s)

Closes #87

_Please be sure to take a look at the [contributing guidelines](../CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](../PR_ETIQUETTE.md)._

juliapampus commented:
> Let's also think about whether we should automate the generation of plantuml files from the artifacts directly to prevent differences between the artifacts and the figures. But that should be in a separate issue.

I've removed the .puml changes as discussed.

arnoweiss reviewed (changes requested):
Took a closer look. The webpage is already building from the artifacts folder as all contents are copied at deploy time [1]. That creates a folder structure like this [2, downloads zip]. So none of the md files need adjustment - currently the figures don't resolve.

I've been thinking about a folder structure in the build that correctly resolves links in respec and github markdown but haven't come up with anything.

[1] https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/blob/8c6c7d6516379242152e526cbef768163d9a76cc/.github/workflows/script.sh#L6
[2] https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/actions/runs/12882779801/artifacts/2460048410

juliapampus commented:
> Took a closer look. The webpage is already building from the artifacts folder as all contents are copied at deploy time [1]. That creates a folder structure like this [2, downloads zip]. So none of the md files need adjustment - currently the figures don't resolve.
> 
> I've been thinking about a folder structure in the build that correctly resolves links in respec and github markdown but haven't come up with anything.
> 
> [1]
> 
> https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/blob/8c6c7d6516379242152e526cbef768163d9a76cc/.github/workflows/script.sh#L6
> 
> 
> [2] https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/actions/runs/12882779801/artifacts/2460048410

Great! Haven't know that. I've just ebased the PR and reverted the .md changes. For the GitHub users, However, I've introduced a `README` file pointing to the `artifacts` folder.
